app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt

```diff
@@ -167,24 +173,35 @@ class BrowserFragment : Fragment(), BackHandler {
 
         startPostponedEnterTransition()
 
-        QuickActionComponent(
+        quickActionSheetStore = Store(
+            QuickActionSheetState(
+                readable = getSessionById()?.readerable ?: false,
+                bookmarked = findBookmarkedURL(getSessionById()),
+                readerActive = getSessionById()?.readerMode ?: false,
+                bounceNeeded = false
+            ),
+            ::quickActionSheetStateReducer
+        )
+
+        val quickActionSheetView = QuickActionView(
             view.nestedScrollQuickAction,
-            ActionBusFactory.get(this),
-            FenixViewModelProvider.create(
-                this,
-                QuickActionViewModel::class.java
-            ) {
-                QuickActionViewModel(
-                    QuickActionState(
-                        readable = getSessionById()?.readerable ?: false,
-                        bookmarked = findBookmarkedURL(getSessionById()),
-                        readerActive = getSessionById()?.readerMode ?: false,
-                        bounceNeeded = false
-                    )
-                )
-            }
+            QuickActionInteractor(
+                context!!,
+                DefaultReaderModeController(readerViewFeature),
+                quickActionSheetStore,
+                shareUrl = ::shareUrl,
+                bookmarkTapped = {
+                    lifecycleScope.launch { bookmarkTapped(it) }
+                }
+            )
         )
 
+        quickActionSheetStore.observe(this) {
+            MainScope().launch {
```

Comment:
We fixed a bug here today in `SearchFragment` which is now using the lifecyclescope: https://github.com/mozilla-mobile/fenix/blob/master/app/src/main/java/org/mozilla/fenix/search/SearchFragment.kt#L170

We're working on better API in A-C: https://github.com/mozilla-mobile/android-components/issues/3803 but for now we need that change. Sorry this is a bit of moving target still.

Reply:
Updated!